### PR TITLE
add gradle dependency checker MAPSAND-341

### PR DESCRIPTION
**Pull request by alexander-kulikovskii**
### Summary of changes
With our dependency organisation Android Studio can't highlight new dependencies. This plugin allows us to check all of them and print in one file.
CI will run this plugin automatically once a week.

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ workflows:
     when:
       not: << pipeline.parameters.mapbox_upstream >>
     jobs:
+      - check-gradle-denendencies
```
Comment (paulrenn67) [inline .circleci/config.yml:22]: typo here "denendencies"

File: .circleci/config.yml [hunk 2]
```diff
@@ -683,6 +684,24 @@ jobs:
                 --target-slug mapbox/mapbox-maps-android-private
             fi
 
+  check-gradle-denendencies:
```
Comment (paulrenn67) [inline .circleci/config.yml:687]: and here

File: .circleci/config.yml [hunk 3]
```diff
@@ -683,6 +684,24 @@ jobs:
...
+    executor:
+      name: ubuntu
+      machine: small
+    steps:
+      - checkout
+      - restore-gradle-cache
+      - run:
+          name: Run dependency check
+          no_output_timeout: 15m
+          command: ./gradlew dependencyUpdates --stacktrace
```
Comment (paulrenn67) [inline .circleci/config.yml:697]: Do we want to keep `--stacktrace` or is it just for testing?

File: .circleci/config.yml (hunk 1 above)
Comment (alexander-kulikovskii) [reply .circleci/config.yml]: fix

File: .circleci/config.yml (hunk 2 above)
Comment (alexander-kulikovskii) [reply .circleci/config.yml]: fix

File: .circleci/config.yml (hunk 3 above)
Comment (alexander-kulikovskii) [reply .circleci/config.yml]: remove it

**Comment by kiryldz** [comment]
@alexander-kulikovskii what does this PR do?

File: build.gradle.kts [hunk 4]
```diff
@@ -27,6 +27,7 @@ buildscript {
     classpath(Plugins.mapboxSdkVersionsPlugin)
     classpath(Plugins.pitestPlugin)
     classpath(Plugins.playPublisher)
+    classpath("com.github.ben-manes:gradle-versions-plugin:0.42.0")
```
Comment (kiryldz) [inline build.gradle.kts:30]: why not put in plugins as all entries above?

Comment (alexander-kulikovskii) [reply build.gradle.kts]: done

**Review by kiryldz — APPROVED** [review]

File: .circleci/config.yml [hunk 5]
```diff
@@ -15,6 +15,17 @@ parameters:
 
 workflows:
   version: 2
+  nightly:
+    triggers:
+      - schedule:
+          # at 7:00 on Monday
```
Comment (jush) [inline .circleci/config.yml:21]: Nit: If it's on Monday's only, should we call it `weekly` instead of `nightly`? Otherwise, it's a bit misleading

**Review by jush — APPROVED** [review]

File: .circleci/config.yml [hunk 6]
```diff
@@ -666,6 +677,28 @@ jobs:
...
+          name: Merge all reports
+          command: python3 scripts/merge-gradle-dependency-output.py
+      - run:
+          name: Zip artifacts
+          command: zip -r results.zip merged_report.txt
+          when: always
+      - store_artifacts:
+          path: results.zip
+          destination: .
+
```
Comment (yunikkk) [inline .circleci/config.yml:701]: Maybe also send the report in slack?

**Pull request merged**